war/pom.xml

```diff
@@ -538,36 +571,67 @@ THE SOFTWARE.
         </plugins>
       </build>
     </profile>
+
+      
+    <!--
+        The following profiles are required to integration the node/gulp build into this maven build.
+        Hopefully we can push these profiles down into a parent pom.
+        See https://github.com/tfennelly/jenkins-js-builder#maven-integration
+    -->
+      
+    <!--
+        Manually download node and NPM for the frontend plugin so it gets cached in the local maven repo.
+        Otherwise frontend will be downloading it all the time on the CI servers. The download-maven-plugin 
```

Comment:
npm modules usually contains different trash and directly downloads from internet that wouldn't be cached.  So if you want really cache everything, this wouldn't help.